### DISCUSSION TITLE
howto stop notification 'AIO is outdated!' for non-admin users ?

### DISCUSSION THREAD

**Discussion by myB0B** [nextcloud/all-in-one#3578] [Questions] [answered]
In a production environment, it is not appropriate for non-admin users to receive notifications from the master container. How can these notifications be disabled?

**Comment by szaimen** [comment]
> howto stop notification 'AIO is outdated!' for non-admin users ? 

Simply update the containers.

**Comment by myB0B** [comment]
Is this not a serious response? We're talking about a production environment with real users, not a geek's home lab in a pimply teenager's bedroom !
Only system admins should receive this kind of notification!

**Reply by szaimen** [reply]
These notifications are only send after 3 months without any update. So if they are sent you did a very bad job with updating the installation.

**Reply by myB0B** [reply]
Updating things is not always the right thing to do, especially in a controlled environment. Sometimes, it takes time to test before updating and sometimes updating is worse, and sometimes it shouldn't be done at all.

3 months? So there is an issue because i updated an AIO instance from Nextcloud 26 to Nextcloud 27 (AIO 6.xx to 7.2.1) on October 1, 2023, and yesterday, I received another update notification!

here is portainer datas (see the date in french time):
`  | nextcloud-aio-mastercontainer | healthy |   | nextcloud_aio | nextcloud/all-in-one:latest | 2023-10-01 08:53:21 | 172.22.0.3 | 7080:8080 `

Please change this behavior; it's really annoying

**Reply by szaimen** [reply]
What defines this id the nextcloud image. Can you post the output of `sudo docker image inspect nextcloud/aio-nextcloud:latest` ?

**Reply by myB0B** [reply]
```
[
    {
        "Id": "sha256:f959816114f65ca18f9fa433a54f25c636754e62604cc985e74b695d167a427e",
        "RepoTags": [
            "nextcloud/aio-nextcloud:latest"
        ],
        "RepoDigests": [
            "nextcloud/aio-nextcloud@sha256:2a4a2b7e5651a5e58498c20b9b5c633a24a0c0d053e801d7753972eea6f79fc7"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2023-09-15T15:04:14.29278299Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "root",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "9000/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev dpkg \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkgconf \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=528995BFEDFBA7191D46839EF9BA0ADA31CBD89E 39B641343D8C104B2B146DC3F9C39DC0B9698544 F1F692238FBC1666E5A5CCD4199F9DFEF6FFBAFD",
                "PHP_VERSION=8.1.23",
                "PHP_URL=https://www.php.net/distributions/php-8.1.23.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.1.23.tar.xz.asc",
                "PHP_SHA256=fc48422fa7e75bb45916fc192a9f9728cb38bb2b5858572c51ea15825326360c",
                "PHP_MEMORY_LIMIT=512M",
                "PHP_UPLOAD_LIMIT=10G",
                "PHP_MAX_TIME=3600",
                "NEXTCLOUD_VERSION=27.1.0",
                "AIO_TOKEN=123456",
                "AIO_URL=localhost"
            ],
            "Cmd": [
                "/usr/bin/supervisord",
                "-c",
                "/supervisord.conf"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "sudo -E -u www-data bash /healthcheck.sh"
                ],
                "StartPeriod": 60000000000
            },
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": {
                "/mnt/ncdata": {},
                "/var/www/html": {}
            },
            "WorkingDir": "/var/www/html",
            "Entrypoint": [
                "/start.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.centurylinklabs.watchtower.enable": "false"
            },
            "StopSignal": "SIGQUIT"
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 929932898,
        "VirtualSize": 929932898,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/62c1d00c465d20ff838a5e3606978feed87dfff2d4551981ae695749aeb9b9b4/diff:/var/lib/docker/overlay2/c3ecb608a25377e6ffd15a4cafbb3237505c44d03a498bdc780828f302b53144/diff:/var/lib/docker/overlay2/0742a7a5d95d6a4cad090e27884f9bdb551d8847faf42e5010c92d66aa67e8db/diff:/var/lib/docker/overlay2/c5221c3ab79aae7e5eb0c8a7f261546f53d67ba68f3ee82deffe6aabc25058b7/diff:/var/lib/docker/overlay2/a76430c4756ae6a6b0d71f6331459180a0fb3504e298fd348a806dd229dabe44/diff:/var/lib/docker/overlay2/81c8931509f0d466988a1dedfd70c4ed03129b79c619f999348b01262a83c4f5/diff:/var/lib/docker/overlay2/ef78ef3f90dbb2eec71b9e164889aa39ae457d776e281d2b11ac67bb6bfc03dc/diff:/var/lib/docker/overlay2/76265ecab8a3044e929101cf063bb1a7fe58c12052e9e8af7804f439266cdfb7/diff:/var/lib/docker/overlay2/7bae652485ab16e42022fa37b52d6527769db0ba30e9f979c6bf1bb0e3c5bcc1/diff:/var/lib/docker/overlay2/7c0350f51ec82863ece5c7a997735d800abd94b3f129a3a25133fefe04a9b571/diff:/var/lib/docker/overlay2/6275ef499de83b05c3cbdf0d8321b3aced8da4589ee682b8f6f79f30dad730d0/diff:/var/lib/docker/overlay2/7ec7f75d5439eb5cc54af05dd2d2a95752f052518fd1c141335e3c0a89ce2af0/diff:/var/lib/docker/overlay2/29c1459e8de51eb243f1b31de5caf1d065ef18d0f06b29086ffe182c3e54735d/diff:/var/lib/docker/overlay2/10f207d3c32e1d37878caeff91e1548dac4f6f38d3793dd9ffcae6e66ecd856d/diff",
                "MergedDir": "/var/lib/docker/overlay2/4c45c36f01c6b6ed1f6a8e42057d3bdce816889cc37325ab1a3b688fdcb410cb/merged",
                "UpperDir": "/var/lib/docker/overlay2/4c45c36f01c6b6ed1f6a8e42057d3bdce816889cc37325ab1a3b688fdcb410cb/diff",
                "WorkDir": "/var/lib/docker/overlay2/4c45c36f01c6b6ed1f6a8e42057d3bdce816889cc37325ab1a3b688fdcb410cb/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:4693057ce2364720d39e57e85a5b8e0bd9ac3573716237736d6470ec5b7b7230",
                "sha256:ebaa87f84e90b753e6fbf6fd7201fb1ffafc33216a534be38c74053fb4e92ef5",
                "sha256:2270494f0a506c069ece8b130f2c3e242065a07564e0bf6363f38ab4b1ecc828",
                "sha256:c11f925ddb3f869c9892eb1224f6c466fde128d045317f0b57ac1fcd510f1f0f",
                "sha256:7888dd7d15b98f18ac99ff1facb89e6b65147018500b9a05835177943d4f3e8d",
                "sha256:3d1a74b1a12c9eccbbb88be104cd6aff8d5c888b18bd08bb3978cf469e3f8f86",
                "sha256:2da9fbc90e2aa96e177fbfbbbf23c02164634274e5946d31d0538d7bb73c36af",
                "sha256:e9a7522c15c6d2952cef7030680dab7b11e16b8415f930bff443a0324d1c4aab",
                "sha256:9b8db7eec999b69a2854fe9ba2138fa727951fd53bd48d318e4266c674ce2369",
                "sha256:a4aee9778681cc2b94ab80686b9ef528b7131eba7f005ea049c647fddbe6c404",
                "sha256:ac6c0f50d316e7966f3fc7a1aa5dfb4fdb9ca850e5b18c22b67982f147567b36",
                "sha256:ff73cca3e86d6b677ae3d20acf2b59507dd4ae64c04fa2f1018d3680b5d148f6",
                "sha256:8c12f5480b6d4300f6801a6e9e97150ce7749ac2d7c3d24fba2770f261b16533",
                "sha256:70dc18abc2402817df8b88dee0f09a636436c19c09137cc345a2b6e1fe395ed9",
                "sha256:0e53c905e454cb9f1c7a870d7c173716827858884bfce5433f7f26cfd3d89802"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

**Comment by szaimen** [comment]
> "Created": "2023-09-15T15:04:14.29278299Z

I agree that in this case the notification should not be sent as 90 days are not over yet. Can you post your mastercontainer logs here? Maybe something related is in there? 

Also when was the last `AIO is outdated` notification sent? Is it still every day?

**Comment by myB0B** [comment]
i update yesterday all my instances ... will see if reminder delay is ok

**Comment by myB0B** [comment]
so after exactly 6 days (3 days ago) there is a new notification for update the master container ! again....
i am disapointed

**Reply by szaimen** [reply]
Was it sent today or is it an old notification? Can yo post a screenshot of the notification here?

**Reply by myB0B** [reply]
![Capture d'écran 2023-11-14 150921](https://github.com/nextcloud/all-in-one/assets/26011945/7c1184fc-4cc9-4c64-afc2-19823617d8a4)

**Reply by szaimen** [reply]
Ah right. This notification is only send once a week and only to admin users. (There are different kind of notifications). So everything seems to work as expected.

**Reply by myB0B** [reply]
ok thanks, It's a bit confusing, there's nothing indicating that it's specifically meant for admin users.

**Reply by pbvdven** [reply]
I have users without any group membership receiving this message every day I deleted all notifications also on there accounts updated to the latest 27 version but still get this outdated notification every morning my admins also but thats not the problem i don't want my users to get this notification and im still having issues with 28 so i don't want to update just jet.

**Comment by deanjarnold** [comment]
@szaimen We also have this issue. All users (non-admins) receive "AIO is outdated!" notification every day.
As stated by other users, in our production environments, we don't always update AIO, we stage update and rollout after testing.

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-often-are-update-notifications-sent

**Comment by mtaipe** [comment]
Got the oudated message so I stopped containers and update them.
A day later got same notification. The Nextcloud AIO Interface says "You are running the latest channel".

**Reply by dralphs** [reply]
Hi @mtaipe, I am experiencing the same thing on Nextcloud Desktop Client Version 3.16.3 (macOS Sequoia 15.3.2) and iOS (version 6.6.0 on iOS 18.4.1).   Did yours clear up?  What clients (and OS) are you using?

**Reply by szaimen** [reply]
Hi, this was a bug and was fixed with v10.13.0

**Reply by Michael-Hennemann** [reply]
> Hi, this was a bug and was fixed with v10.13.0

Did an AIO update yesterday (Jan. 7th), today I get the "AIO is outdated" message  
Cheers
Michael

**Comment by dralphs** [comment]
Hi Simon, thank you for the reply!

I've updated my containers and will keep an eye out for any odd behavior.

Thank you, 

David

**Comment by paulwedeck** [comment]
Hello,
would it be possible to at least allow non-admin users to opt out or at least temporarily block the message?
I use a nextcloud instance which is behind on updates (current version from end of April) which is not my fault but I have to read this message daily on several devices.
Regards
Paul

**Comment by szaimen** [comment, marked as answer]
Hi, I reconsidered this feature and we will sent the notification in the future only to admin users. See https://github.com/nextcloud/all-in-one/pull/8577

**Reply by szaimen** [reply]
This is now released with v13.6.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
